packages/core/core/src/public/BundleGraph.js

```diff
@@ -133,11 +134,24 @@ export class MutableBundleGraph extends BaseBundleGraph
   }
 
   addBundleGroup(parentBundle: ?IBundle, bundleGroup: BundleGroup) {
-    let node = {
-      id: getBundleGroupId(bundleGroup),
-      type: 'bundle_group',
-      value: bundleGroup
-    };
+    let bundleGroupId = getBundleGroupId(bundleGroup);
```

Comment:
This implements bundle group "sharing": if a node for this group already exists, we reference it instead. However, the `dependency` on the group will continue to reference the first bundle that depends on it. We should probably decouple dependencies from bundle groups, and store those connections in the bundle graph instead.

Reply:
What do we use the `dependency` for?

Reply:
We use them

* When making new connections when adding a bundle to a group ([0] and [1])
* in the JS Runtime to determine if the bundle group is brought in via a URL or not (whether to add a runtime or not) [2]

[0] https://github.com/parcel-bundler/parcel/blob/v2/packages/core/core/src/public/BundleGraph.js#L146
[1] https://github.com/parcel-bundler/parcel/blob/v2/packages/core/core/src/public/BundleGraph.js#L186
[2] https://github.com/parcel-bundler/parcel/blob/v2/packages/runtimes/js/src/JSRuntime.js#L83